linux/keyman-config/install_kmp.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/python3
+
+import argparse
+import json
+import os.path
+import sys
+import tempfile
+import zipfile
+from os import listdir, makedirs
+from shutil import copy2
+
+import requests
+
+from get_kmp import get_keyboard_data, get_kmp
+from kmpmetadata import determine_filetype, parseinfdata, parsemetadata
+from list_installed_kmp import get_kmp_version
+from uninstall_kmp import uninstall_kmp
+
+
+def list_files(directory, extension):
+	return (f for f in listdir(directory) if f.endswith('.' + extension))
+
+def extract_kmp(kmpfile, directory):
+	with zipfile.ZipFile(kmpfile,"r") as zip_ref:
+		zip_ref.extractall(directory)
+
+def get_metadata(tmpdirname):
+	"""
+	Get metadata from kmp.json if it exists.
+
+	Args:
+		inputfile (str): path to kmp file
+		tmpdirname(str): temp directory to extract kmp
+
+	Returns:
+		list[5]: info, system, options, keyboards, files
+			see kmpmetadata.parsemetadata for details
+	"""
+	kmpjson = os.path.join(tmpdirname, "kmp.json")
+	if os.path.isfile(kmpjson):
+		return parsemetadata(kmpjson, False)
+	else:
+		return None, None, None, None, None
+
+def get_infdata(tmpdirname):
```

Comment:
I think you can remove the kmp.inf processing. That's primarily for Windows installations, and kmp.json is the preferred metadata file.

Reply:
There are still some kmp (at least were when I wrote that) that only have kmp.inf and no kmp.json